data-munging/divest_spd_links.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+from pathlib import Path
+
+import click
+import pandas as pd
+
+
+def main(id_path: Path, link_path: Path, output: Path):
+    ids = pd.read_csv(id_path, usecols=["id", "badge number"])
+    links = pd.read_csv(link_path)
+    # Collapse the multiple links columns into a single column list
+    links["url"] = links[
+        ["Links", "last updated: 8/6/2021", "Unnamed: 6"]
+    ].values.tolist()
+    # Make the badge column a string
+    links["badge number"] = links["Badge Number"].astype(str)
+    # Subset the columns at this point
+    links = links[["badge number", "url"]]
+    # Explode the list of links into individual rows
+    links = links.explode("url", ignore_index=True)
+    # Drop any rows where a link doesn't exist (this will be most)
+    links = links[links["url"].notna()]
+    # Join the two dataframes on badge number, discard any rows with missing values
+    merged = links.merge(ids, how="inner").astype({"id": pd.Int64Dtype()})[
+        ["id", "url"]
```

Comment:
`pandas` is magic to me, I have no idea what any of these incantations mean.

Reply:
Ha, it's **the** tabular data manipulation tool for python. And FWIW this is probably the easiest package of its ilk to understand :sweat_smile: The ones I've seen in other languages are even more abstruse! Also that's why I leave comments for every line lol, I'm sure I'll forget the meaning of some of these steps down the line.

A lot of the data-specific PRs are going to involve some heavy panda usage, apologies that there'll likely be more of this coming your way haha